docs/ballot_encoding.md

```diff
@@ -0,0 +1,67 @@
+# Encoding of a ballot
+
+
+Here is described a common enconding for ballots, which is needed
+to make sure that all ballots can be encoded in a unique way (up to the
+ordering of questions).
+
+## Answers
+The answers to questions are encoded in the following way:
+
+```
+type:ID:<answers>
+```
+Where we have that:
+* the type can either be "select", "rank" or "text"  
+* ID is the ID of the question 
+* The 'answers' are seperated by commas
```

Comment:
I would try to use some sort of metasyntax (https://en.wikipedia.org/wiki/Metasyntax). For example:

```
<type><sep><id><sep><answers>

TYPE = "select"|"text"|"rank"
SEP = ":"
ID = int8
ANSWERS = <answer>[","<answer>]*
ANSWER = <select_answer>|<text_answer>|<rank_answer>
SELECT_ANSWER = int8
...
```